pkg/util/digest/digest.go

```diff
@@ -108,10 +108,16 @@ func ComputeForIntegration(integration *v1.Integration, configmaps []*corev1.Con
 	// Calculation logic prior to 1.10.0 (the new Traits API schema) is maintained
 	// in order to keep consistency in the digest calculated from the same set of
 	// Trait configurations for backward compatibility.
-	traitsMap, err := toMap(integration.Spec.Traits)
+	traits := integration.Spec.Traits.DeepCopy()
+	// Merge with executed traits
+	if err := traits.Merge(integration.Status.Traits); err != nil {
```

Comment:
I may have missed something but looks like here we are overriding spec traits wit status traits ? at least from this https://github.com/apache/camel-k/blob/7dfac64564a85c5f375924e67901a13fbfd1ce30/pkg/apis/camel/v1/common_types_support.go#L67 method 

is this what we expect/correct ?

Reply:
Yes. The idea is to evaluate the digest of both spec traits and executed traits in order to validate any possible difference between phase transitions. Ie, moving from a phase to another, the result of traits execution may be different and has to be taken in consideration.

Reply:
yes but if the status overrides the spec, the it may not detect drift, I am wrong ?

Reply:
thinking a little bit more, you may be want to compute the checksum for the trais defined in the spec and those in the status, not the merge otherwise you may always hit the issue that one side would hide a possible drift.

Reply:
I think the merge behavior is to replace those properties which are in both data structures. In our case, the status ones are meant to replace the spec ones. I've provided a test (only to calculate the digest) in order to prove it. I think the logic is fine by covering both cases. Please, have a look at the amended commit.

Reply:
So what I'm talking about is that, if a spec trait value changes, and the same trait is set in the status, then the digest won't change so the test below, fails:

```go
func TestSpecStatusDrift(t *testing.T) {
	it := v1.Integration{}
	it.Spec.Traits.Camel = &trait.CamelTrait{}
	it.Status.Traits.Camel = &trait.CamelTrait{}

	it.Spec.Traits.Camel.Properties = []string{"hello=world1"}
	d1, _ := ComputeForIntegration(&it, nil, nil)

	it.Status.Traits.Camel.Properties = []string{"hello=world2"}
	d2, _ := ComputeForIntegration(&it, nil, nil)

	it.Spec.Traits.Camel.Properties = []string{"hello=world3"}
	d3, _ := ComputeForIntegration(&it, nil, nil)

	assert.NotEqual(t, d2, d1, "d2 must not be equal to d1")
	assert.NotEqual(t, d3, d2, "d3 must not be equal to d2")
}
```

```
=== RUN   TestSpecStatusDrift
    digest_test.go:250: 
        	Error Trace:	/home/luca/work/dev/camel-k/pkg/util/digest/digest_test.go:250
        	Error:      	Should not be: "vZ9xT9oBngdDxEgHDEWFW2xqeq-rhHJK2zi-qNruAf-U"
        	Test:       	TestSpecStatusDrift
        	Messages:   	d3 must not be equal to d2
--- FAIL: TestSpecStatusDrift (0.00s)
```

Reply:
True. We probably need to address both values separately in order to catch the change. I've amended the commit to do so and with your permission I've included the test you've written above. I think this should solve the problem now.